design/ceph/object/ceph-sse-s3.md

```diff
@@ -0,0 +1,44 @@
+---
+title: include aws server side encryption s3 support for RGW 
+target-version: release-1.10
+---
+
+# Feature Name
+AWS server side encryption S3 support for RGW
+
+## Summary
+The S3 protocol over three different types of [server side encryption](https://docs.aws.amazon.com/AmazonS3/latest/userguide/serv-side-encryption.html): SSE-C, SSE-KMS and SSE-S3. For the last two RGW server need to configure with external services such as `vault`. Currently Rook can configure RGW with `SSE-KMS` options, so that S3 request with `sse:kms` header will be handled properly. Recently the support for handling the `sse:s3` added to RGW with another set of options, so Rook needs to provide option to configure RGW with those options.
+
+### Goals
+Configure RGW with `SSE-S3` options, so that RGW can handle request with `sse:s3` headers.
+ 
+### Non-Goals
+Both `SSE-KMS` and `SSE-S3` will be configured with same set of values such as `vault-server`, `vault-secret-engine` etc
+
+## Proposal details
+Introducing new field `ServerSideEncryptionSpec` in `CephObjectStoreSpec` which defines types of `AWS-SSE` configured with RGW.
+```
+type AWSServerSideEncryptionSpec {
+	S3Enabled bool `json:"s3Enabled,omitempty"`
+	KMSEnabled bool `json:"kmsEnabled,omitempty"`
+}
+type RGWServerSideEncryptionSpec struct {
+	AWS AWSServerSideEncryptionSpec `json:"aws,omitempty"`
+}
+type ObjectStoreSpec struct {
+...
+	ServerSideEncryption RGWServerSideEncryptionSpec `json:"serverSideEncryption,omitempty"`
+...
+} 
+```
+The values for this configuration will be available in [Security field](Documentation/CRDs/Object-Storage/ceph-object-store-crd.md#security-settings) of `CephObjectStoreSpec`, so depending on the above option RGW can be configured with `SSE-KMS` and `SSE-S3` options. If `ServerSideEncryption` is empty but `Security` is not empty then RGW will be configured with `SSE-KMS` options for the backward compatibility. 
+
+## Alternatives
```

Comment:
I think the point is to show what toolbox commands will be executed by the operator, right?
```suggestion
## Config Commands
```